mne_bids/path.py

```diff
@@ -958,32 +958,21 @@ def update(self, *, check=None, **kwargs):
                     f"Key must be one of " f"{ALLOWED_PATH_ENTITIES}, got {key}"
                 )
 
+            if isinstance(val, int):
+                kwargs[key] = f"{val}"
+
             if ENTITY_VALUE_TYPE[key] == "label":
                 _validate_type(val, types=(None, str), item_name=key)
             else:
                 assert ENTITY_VALUE_TYPE[key] == "index"
                 _validate_type(val, types=(int, str, None), item_name=key)
                 if isinstance(val, str) and not val.isdigit():
                     raise ValueError(f"{key} is not an index (Got {val})")
-                elif isinstance(val, int):
-                    kwargs[key] = f"{val:02}"
 
         # ensure extension starts with a '.'
         extension = kwargs.get("extension")
         if extension is not None and not extension.startswith("."):
-            warn(
-                f'extension should start with a period ".", but got: '
-                f'"{extension}". Prepending "." to form: ".{extension}". '
-                f"This will raise an exception starting with MNE-BIDS 0.12.",
-                category=FutureWarning,
-            )
             kwargs["extension"] = f".{extension}"
-            # Uncomment in 0.12, and remove above code:
-            #
-            # raise ValueError(
-            #     f'Extension must start wie a period ".", but got: '
-            #     f'{extension}'
-            # )
```

Comment:
can you see in the PR tha introduced this what was the rationale for this warning? I am sure it was discussed.

Reply:
According to the BIDS specs, extensions start with a period, see e.g.
https://bids-specification.readthedocs.io/en/stable/appendices/meg-file-formats.html#neuromagelektamegin

I'm therefore -1 on this change for now. It's a tradeoff between usability and following the standard closely…

Reply:
Thanks for sharing that. This is where I get really confused on BIDS and the interaction with the user. I think that is an important technical distinction for comparing across packages perhaps but other than that I really fail to see how that is relevant to an end user. Leading period, no leading period, the convention is clear that all file names will have a period before the extension whether it's in the extension or not. So why are we torturing the user by making them consider this minutia when it doesn't have any functional difference? I'll look at the PR for more reasoning but when I was learning BIDS, I spent about 10x the time learning about this irrelevant stuff really and only 1x learning about crucial sidecar files and their required and recommended fields. And to be honest, it was a terrible learning experience, I still often feel lost in all the different style recommendations and requirements (see for instance this zero padding) and I've been using and developing BIDS for years! To be even more honest, I don't actually use `mne_bids` to convert my datasets anymore, I pulled out and simplified each sidecar function generator and I write each sidecar at the appropriate time (i.e. I write events after I find the events, I write channels after I find the bad channels and I write electrodes after I find the positions), I find that way more intuitive and then I stop and helps me think about what's needed in each step and each sidecar. That just works much better for me and my workflow, I'm not saying it's better for everyone but these types of things drive me away from using the `mne_bids` API because it's such a simple and practice fix to just fix one error in the ten you might run into if you've never converted a dataset before and I think a lot of rationale people stop at 9 or less!

Reply:
Good points! So the idea is to add a period if none was supplied? Yes, why not! I didn't fully think this through before posting my comment above 

Could you perhaps describe your custom workflow in more detail?

Like I said elsewhere, I, too, believe the API is difficult to master and very inflexible. If we could divide the workflow into more small, independent sub-steps, things might become easier to grasp

Reply:
No worries, I'm just trying to be transparent and giving my 2 cents, I think it's very reasonable to default to the specification and I think you're doing a good job with the code I've seen you change, I just made this PR because I was tired of running into little errors like this.

For intracranial recordings, I find the events from a photodiode channel then save those to events, I find the contacts from a post op CT and save those to electrodes and I find the bad channels and save those to channels, not much more to it than that!

Reply:
I like the idea of allowing for a more "incremental" writing if the BIDS data. Like, maybe write the raw file right away after recording; update the events later after manual inspection; etc.

@sappelhoff WDYT? I'd be happy to brainstorm about this in a video call or the next time I come to Berlin :)